Handle passwords with special characters in FTP Plugin

ndouthit opened the pull request:
Expanding on #25 and the FTP Plugin from #113.

If your username or password includes any special characters as part of the FTP URI, an exception is thrown. The username/password need to be properly encoded to be used in a URI. If your password is _P@ssw0rd!_ it would need to be entered as _P%40ssw0rd!_ in order for the URI to be created properly. However, if you encode the password this way, your credentials will be rejected because _P%40ssw0rd!_ is sent to the FTP server instead of _P@ssw0rd!_.

This pull request separates the username and password from the FTP URI which allows for the use of special characters. It also stores the password in a SecureString instead of a string.

ndouthit commented:
I also noticed an issue where the .well-known and acme-challenge directories were not being properly deleted so I threw in a fix for that.

In addition I noticed that the .well-known and acme-challenge directories are not created automatically if they do not exist. I haven't implemented a fix for that, but it will be necessary if you plan on deleting the answer file and the directories.

rkerber commented:
There were a couple of issues I fixed in my latest commit.
I had some errors pop up in my testing, and I added the ability for it to create directories if they don't exist.

ndouthit commented:
Awesome. Looks like the merge accidentally includes some code that was
meant for testing.

On Saturday, February 13, 2016, Robert Kerber <notifications@github.com>
wrote:

> There were a couple of issues I fixed in my latest commit.
> I had some errors pop up in my testing, and I added the ability for it to
> create directories if they don't exist.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/Lone-Coder/letsencrypt-win-simple/pull/123#issuecomment-183811275>
> .
>

rkerber commented:
Thanks for reminding me, I forgot to delete that. Good thing the password is a secure string so I couldn't put it in there :)
